SECEdgar/filings/base.py

```diff
@@ -43,75 +44,98 @@ def dateb(self, val):
                 type will be returned.
         """
         if not isinstance(filing_type, FilingType):
             raise FilingTypeError(FilingType)
 
         return filing_type
 
-    def _validate_cik(self, cik):
+    @staticmethod
+    def _validate_cik(cik):
         """Validates that given CIK *could* be valid.
 
         Args:
-            cik (Union[str, int]): Central index key (CIK) to validate.
+            cik (Union[CIK, str, int]): Central index key (CIK) to validate.
 
         Returns:
-            cik (Union[str, int]): Validated CIK.
+            cik (Union[str, list of str]): Validated CIK.
                 Note that the CIK is only validated in
-                that it *could* be valid. All CIKs
-                must be 10 digits, but not all 10 digit
+                that it *could* be valid. CIKs formatted as
+                10 digits, but not all 10 digit
                 numbers are valid CIKs.
 
         Raises:
-            ValueError: If given cik is not str or int
-            CIKError: If cik is not a 10 digit number.
+            ValueError: If given cik is not str, int, or CIK object.
+            CIKError: If cik is not a 10 digit number or valid CIK object
         """
-        if not isinstance(cik, (str, int)):
-            raise ValueError("CIK must be of type str or int.")
-        elif isinstance(cik, str):
-            if len(cik) != 10 or not cik.isdigit():
-                raise CIKError(cik)
-        elif isinstance(cik, int):
-            if cik > 10**10:
-                raise CIKError(cik)
-            elif cik < 10**9:
-                return str(cik).zfill(10)  # pad with zeros if less than 10 digits given
-        return cik
+        # creating CIK object should check to see if ciks are valid
+        if not isinstance(cik, CIK):
+            if not isinstance(cik, (str, int)):
+                raise ValueError("CIK must be of type str or int.")
+            elif isinstance(cik, str):
+                if len(cik) != 10 or not cik.isdigit():
+                    raise CIKError(cik)
+            elif isinstance(cik, int):
+                if cik > 10**10:
+                    raise CIKError(cik)
+                elif cik < 10**9:
+                    return str(cik).zfill(10)  # pad with zeros if less than 10 digits given
+            return str(cik)
+        else:
+            return cik.cik
 
     def _get_urls(self):
-        """Get urls for txt files.
+        """Get urls for all CIKs given to Filing object.
 
         Returns:
             urls (list): List of urls for txt files to download.
         """
-        url = self._prepare_query()
-        data = self._execute_query(url)
+        if isinstance(self._cik, (list, tuple, set)):
+            urls = list()
+            for cik in self._cik:
+                urls.append(self._get_cik_urls(cik))
+            return urls
+        else:
+            return self._get_cik_urls(self._cik)
+
+    def _get_cik_urls(self, cik):
```

Comment:
Why not shift this responsibility to `CIK`?

Reply:
Good idea. We will need the filing type. Will this be the only parameter passed to the `_get_urls` method?

Reply:
I guess for the current implementation, it is fine, we can pass the CIK.



In an ideal world, we can have independent NetworkClient then we can use it here also. In this case, network client takes the responsibility of creating the URLs and fetching them and so the CIK class might look like this.

```
class CIK(object):
    ___init__(network):
     self.network = network

    def get_url_for(cik):
       self.network.fetch_cik_code(company_name)
```